### DISCUSSION TITLE
[Mesh] Meshing a small cubic within a big cubic

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#24158] [Q&A Meshing] [answered]
Dear all,

I want to use MOOSE Mesh blocks to build a small cubic (i.e., 2m*2m*2m) within a big cubic (i.e., 1km*1km*1km).

One solution may be to build a quite refined mesh (i.e., 0.25m*0.25m*0.25m) for the whole big cubic (i.e., 1km*1km*1km) and then use [SubdomainBoundingBoxGenerator](https://mooseframework.inl.gov/source/meshgenerators/SubdomainBoundingBoxGenerator.html) to select the small cubic. But it needs many elements for the out parts of the small cubic. I don't want this.

In addition, I try to build and refine the two cubics by 'FileMeshGenerator' and then using 'StitchedMeshGenerator' to merge them together, but it produces repeated meshing in their intersection parts.

Could you please give me some suggestions on this? Thanks a lot.

J

**Comment by GiudGiud** [comment, marked as answer]
Hello

the solution here is likely to use mesh refinement. 

First : generate the uniform grid for the whole domain. It should be as coarse as you want the large coarse part to be. CartesianMG or rectilinearMG should both be able to do it

second: use the subdomain bounding box to create a different subdomain for the fine part. You could skip that part if you use the subdomain_ids argument of the rectilinearMG

Then use the BlockRefineGenerator to refine only the fine part a few levels. You want to get to the fineness required there

finally, if the fine part was “too big” in step 2, you can redefine it again using another subsomainBoundingBox generator

Guillaume

**Reply by Joseph-0123** [reply]
> Hello
> 
> the solution here is likely to use mesh refinement.
> 
> First : generate the uniform grid for the whole domain. It should be as coarse as you want the large coarse part to be. CartesianMG or rectilinearMG should both be able to do it
> 
> second: use the subdomain bounding box to create a different subdomain for the fine part. You could skip that part if you use the subdomain_ids argument of the rectilinearMG
> 
> Then use the BlockRefineGenerator to refine only the fine part a few levels. You want to get to the fineness required there
> 
> finally, if the fine part was “too big” in step 2, you can redefine it again using another subsomainBoundingBox generator
> 
> Guillaume

Thanks a lot. I will try this way.
